### ISSUE TITLE
How long does it take for unconfirmed transactions to become orphaned in the wallet?

### ISSUE THREAD

**Issue by educatedwarrior** [stablecoin/stablecoin#6]
I have some transactions that were sent out around the 12/23/13 in my wallet.   They were not confirmed due to the fork which occurred around that time.  When do the they become orphaned and the funds appear back in the wallet?

**Comment by interhive** [comment]
If you re-download the blockchain, you should see those transactions show up in your wallet, as you'll be on the v1.5 chain.

**Comment by educatedwarrior** [comment]
Been there ,  done that.   These transactions were sent on the wrong fork and so they should be invalidated in the wallet and the funds returned to the wallet.     In other coin clients immature transactions that have not been confirmed are usually orphaned and the funds returned to the wallet.

**Comment by interhive** [comment]
You may want to run a rescan on your wallet. This will rescan the blockchain and rebuild your balance.

I don't think you can do it from the debug window... have to start your client with the `-rescan` flag. If you're running a Qt version, create a shortcut that includes the `-rescan` flag after the .exe.

Example `stablecoin-qt -rescan`

Hope this helps.

**Comment by educatedwarrior** [comment]
I gave your suggestion a try and that didn't work as well.  There may be any issue with the StableCoin wallet on how it handles orphaned/invalid transactions.   If transactions are not confirmed in a period of time, they should be invalidated but this is not happening.  I have a lot of transactions with "?" .   Any plans to put out an updated to fix this?

**Comment by educatedwarrior** [comment]
If you could fix this I will give you a donation.  I have a lot of money tied up in this issue.

**Comment by educatedwarrior** [comment]
Hi interhive.  Just following up.  Any plans to add a feature to orphan transactions after they haven't been confirmed in so many blocks or time?  I got a lot of coins in my wallet tied up in this last fork that occured.   Would appreciate knowing what your plans are either way.

**Comment by interhive** [comment]
@educatedwarrior sorry... I'm not a developer on this project. I was simply hoping to help with some steps I would take if I were in the same boat.

I think I'd try making a backup of the wallet and exporting the public/private key of each of your addresses. I would blow away all files associated with your stablecoin install (%APPDATA/StableCoin I believe on Windows or ~/Library/Application Support/Stablecoin on Mac). Since you have a backup of your wallet.dat, you should be alright. 

Then, with a fresh data directory, import the private key into the new install using the `importprivkey` command. You'll probably want to add the StableCoin Block Explorer (sbc.blockexplorer.io) node to your new config to ensure that you're getting the official blockchain:

```
addnode=66.228.60.13
```

You can reach the console using the Debug option under the Help menu, I believe. I'm sorry you're having these issues.

**Comment by educatedwarrior** [comment]
@interhive , thanks for your suggestion but I already did all that.   I was assuming anyone I talked to on github was a developer of the coin.   Who can I speak to about this issue - I believe it's an issue with the code of the coin and not the local data I have on my machine.   

I simply have thousands of stablecoins in my wallet that have not been sent from my wallet, and I can't use them because the transaction was never orphaned.    This is a serious bug and I need to talk to a developer.

**Comment by interhive** [comment]
@educatedwarrior if these transactions truly occurred on a forked blockchain, they will not be there in the true blockchain. I'm sure you've done this but have you verified that your balance is correct by searching the address on the new blockchain @ http://sbc.blockexplorer.io?

If you've blown away all the app data (including wallet.dat), ensured that you're running version 1.5 of the client, downloaded the true blockchain and imported the private keys associated with your addresses, I just don't see how you'd see orphan transactions in your stablecoin install. It just doesn't make sense.

I suppose your only other option is to transport the private keys to another computer and load them on a separate stablecoin installation. Once they load fine, send them to another address that you've created.

I'm sorry that you're having issues. Drop me a screenshot of what you're seeing to chris@litecoin.io.

**Comment by educatedwarrior** [comment]
@interhive ,  this is a screen shot I took today of transactions that have been sent out over a month ago.   They are still sitting in my wallet unconfirmed because they were sent out a forked blockchain.  The never became orphaned or invalid even after I rebuilt the blockchain db on the correct fork.

![image](https://f.cloud.github.com/assets/4054610/1954895/167efe2a-81e0-11e3-809c-ab9d8013c991.png)

Basically, what I think needs to be done is transactions that have not been confirmed in a period of time or change in height of the blockchain, they should be deleted from the wallet.    The coins that were mined are in the wallet, they just cannot be used because of these false transactions.
